### PR TITLE
[ProjectX] Do not depend on HZ snapshot

### PR DESCRIPTION
**Pull request by olukas**
This PR adds tooling for not to depend on latest HZ IMDG snapshot but rather on some particular version.

Checklist
- [ ] Tags Set
- [ ] Milestone Set
- [ ] Any breaking changes are documented
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] For code samples, code sample main readme is updated

### CODE REVIEW

File: update-snapshot-version.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+git checkout -b ${BRANCH_NAME}
+
+# change version
+echo "Starting change version section"
+sed -i "s#<hazelcast.version>.*</hazelcast.version>#<hazelcast.version>${HAZELCAST_LATEST_VERSION}</hazelcast.version>#" pom.xml
+sed -i "s#<hazelcast.ee.version>.*</hazelcast.ee.version>#<hazelcast.ee.version>${HAZELCAST_EE_LATEST_VERSION}</hazelcast.ee.version>#" pom.xml
+
+# try to compile
+echo "Starting try to compile section"
+mvn clean install -DskipTests -Dmaven.repo.local=${MVN_REPOSITORY}
```
Comment (viliam-durina) [inline update-snapshot-version.sh]: I would leave this for the PR builder, no need to wait for it locally...

File: update-snapshot-version.sh [hunk 2]
```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+JET_EE_LOCAL_PATH=$1
```
Comment (viliam-durina) [inline update-snapshot-version.sh]: We should fail if parameter is missing

File: update-snapshot-version.sh [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+HAZELCAST_SNAPSHOT_VERSION=4.0-SNAPSHOT
+USER_GIT_REMOTE_REPO_NAME=origin
+MVN_REPOSITORY=~/.m2/repository
+ORIGINAL_PATH=`pwd`
```
Comment (viliam-durina) [inline update-snapshot-version.sh]: `dirname $0` works also if we're not running the script from its directory

**Comment by olukas** [comment]
@viliam-durina it's still WIP version. There are a lot of things which need to be improved etc. Since 4.0 is almost done I plan to improve it after release and start to use it for 4.1. (It will also cause some changes in release scripts etc.)

**Comment by viliam-durina** [comment]
We'll depend on the released `4.0` for as long as we can...

**Pull request closed without merging**